test/native/integration-test-helpers/wait-for-store-resolvers.js

```diff
@@ -25,7 +25,7 @@ export async function waitForStoreResolvers( fn ) {
 		const result = fn();
 
 		// Advance all timers allowing store resolvers to resolve.
-		act( () => jest.runAllTimers() );
+		act( () => jest.runOnlyPendingTimers() );
```

Comment:
Out of curiosity why is this change needed?

Reply:
I do not fully know. 😬 

For future reference, I often capture nuanced details in individual commit messages. I try to remember to surface these in pull request descriptions and comments, but I often fail to remember that step. Sorry about that. 

For this change, I captured the "why" in the body of https://github.com/WordPress/gutenberg/pull/50293/commits/34d6d78ddb17889fb69c3438eab8b425b834847c, which I will paste below for convenience. 

It would appear that inner blocks — definitely Social Links, but likely others — result in a scenario where there are _a lot_ of set timers. To my understanding, [running "pending" timers](https://jestjs.io/docs/jest-object#jestrunonlypendingtimers) means Jest runs only the _currently pending_ timers already queued at a given moment, but it will not run additional timers that are queued as a result of running that first queue of timers, i.e. chained timers. I'm unsure where these chained timers occur. Scoping the timers to "pending" resolved the test timeout errors. 

My guest is that rendering each inner block — which Social Links render several to start — generates additional resolvers or other timers. 

Do you think it is worth pursuing a root cause further? Is it worth leaving an inline comment? 

---

Running all timers could create testing failures in scenarios wheres
where inner blocks were inserted, e.g. Social Icons. It is not known
why, but scoping from "all" to "pending" timers resolved the following
errors, without breaking any existing tests.

Unknown origin:

```
Warning: You seem to have overlapping act() calls, this is not supported. Be sure to await previous act() calls before making a new one.
```

Originating from `waitForStoreResolvers`:

```
Ran 100000 timers, and there are still more! Assuming we've hit an infinite recursion and bailing out...
```

Originating from `packages/block-editor/src/components/block-list/test/index.native.js`:

```
thrown: "Exceeded timeout of 5000 ms for a test.
```

Reply:
> For future reference, I often capture nuanced details in individual commit messages. I try to remember to surface these in pull request descriptions and comments

I will remember that thanks! I do think is still worth mentioning something in the description and maybe referencing to the commit message.

> It would appear that inner blocks — definitely Social Links, but likely others — result in a scenario where there are a lot of set timers. To my understanding, [running "pending" timers](https://jestjs.io/docs/jest-object#jestrunonlypendingtimers) means Jest runs only the currently pending timers already queued at a given moment, but it will not run additional timers that are queued as a result of running that first queue of timers, i.e. chained timers. I'm unsure where these chained timers occur. Scoping the timers to "pending" resolved the test timeout errors.

> My guest is that rendering each inner block — which Social Links render several to start — generates additional resolvers or other timers.

I see! I wanted to connect the dots and now I understand that is related to the updated test that now adds a Social links block which surfaced this issue.

> Do you think it is worth pursuing a root cause further? Is it worth leaving an inline comment?

I don't think so, there will be a commit reference anyways for this change and this discussion. 👍 

> Running all timers could create testing failures in scenarios wheres
where inner blocks were inserted, e.g. Social Icons. It is not known
why, but scoping from "all" to "pending" timers resolved the following
errors, without breaking any existing tests.

Oh, I have seen the `Exceeded timeout of 5000 ms for a test` message before, I'm happy this solves this one and the other ones.

This change looks good to me, thanks for explaining!

Reply:
I haven't checked further this case, but based on the findings I shared [here](https://github.com/wordpress-mobile/gutenberg-mobile/issues/5742#issuecomment-1538715579), it's likely that the infinite recursion mentioned above is caused by some component using the `debounce`  function.